unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs

```diff
@@ -55,9 +55,11 @@ public void UpdateData(UserProfileModel newModel)
     {
         if (newModel == null)
         {
-            model = null;
+            model = new UserProfileModel();
```

Comment:
This line feels really wrong. Could you explain what is happening here? Why are we creating a new model? Isn't line 47 in charge of this? Also, shouldn't we check if `model` is not-null before assigning a new model to it?

Reply:
If the current model is null, like it was in the old code, it will throw exceptions when trying to update it again. Im instantiating it again so its a valid object with default values, so the flow doesnt break. Line 47 is just an initialization, but it was never getting instantiated again after nulling the model.